go-testing-tutorial/main_test.go

```diff
@@ -138,3 +138,40 @@ func TestHttp(t *testing.T) {
 		t.Error("Test failed. Expected status to equal expected service response.")
 	}
 }
+
+func TestSearchHandler(t *testing.T) {
+	handler := func(w http.ResponseWriter, r *http.Request) {
+		// here we write our expected response, in this case, we return a
+		// JSON string which is typical when dealing with REST APIs
+		io.WriteString(w, "{ \"status\": \"expected service response\"}")
+
+		param1 := r.URL.Query().Get("search")
+		if param1 != "" {
+			w.WriteHeader(400)
+		} else {
+			w.WriteHeader(200)
+		}
+	}
+
+	req, err := http.NewRequest("GET", "/?search=Testing", nil)
+	if err != nil {
+		return
+	}
+	w := httptest.NewRecorder()
+	handler(w, req)
+
+	resp := w.Result()
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	if resp.StatusCode == 400 {
```

Comment:
Above you supply a `search` query string parameter, so I this check should be 200, or if you meant to test for a 400, you need to remove the `search` query string parameter above (the handler code also looks backwards from what you want.

Reply:
The logic is still backwards here. This test is supposed to fail if the response status code is not 400 (given the lack of `search` parameter)